src/rust/ide/ast/impl/src/crumbs.rs

```diff
@@ -247,6 +267,64 @@ pub enum DefCrumb {
 }
 
 
+// === Match ===
+
+#[allow(missing_docs)]
+#[derive(Clone,Debug,PartialEq,Eq,Hash,PartialOrd,Ord)]
+pub enum MatchCrumb {
+    Pfx  {val:Vec<PatternMatchCrumb>        },
+    Segs {val:SegmentMatchCrumb, index:usize},
+}
+
+#[allow(missing_docs)]
+#[derive(Clone,Copy,Debug,PartialEq,Eq,Hash,PartialOrd,Ord)]
+pub enum PatternMatchCrumb {
+    Begin,
+    End,
+    Nothing,
+    Build,
+    Err,
+    Tok,
+    Blank,
+    Var,
+    Cons,
+    Opr,
+    Mod,
+    Num,
+    Text,
+    Block,
+    Macro,
+    Invalid,
+    Except,
+    Tag,
+    Cls,
+    Or,
+    Seq  {right:bool },
```

Comment:
`seq.which==Left`  is so much nicer than `seq.right==false`.  "not right" is a very cumbersome way of expressing "left".
This is not a flag value, this is enum which happens to have two elements. 
Also, consistency with other code, like `Switch`.